File: DATN-Fashion-Shop-BE/src/main/java/com/example/DATN_Fashion_Shop_BE/dto/response/BaseResponse.java
package com.example.DATN_Fashion_Shop_BE.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.MappedSuperclass;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Data//toString
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@MappedSuperclass
public class BaseResponse {
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss", shape = JsonFormat.Shape.STRING)
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss", shape = JsonFormat.Shape.STRING)
    private LocalDateTime updatedAt;

    private Long createdBy;

    private Long updatedBy;
}

File: FE/DATN-Fashion-Shop-FE/src/app/dto/Response/base-response.ts
export class BaseResponse {
  createdAt: Date;
  updatedAt: Date;
  createdBy?: number;
  updatedBy?: number;

  constructor(data?: Partial<BaseResponse>) {
    this.createdAt = data?.createdAt ? new Date(data.createdAt) : new Date();
    this.updatedAt = data?.updatedAt ? new Date(data.updatedAt) : new Date();
    this.createdBy = data?.createdBy ?? undefined;
    this.updatedBy = data?.updatedBy ?? undefined;
  }
}
